src/test/java/com/faforever/api/clan/ClanServiceTest.java

```diff
@@ -0,0 +1,334 @@
+import org.springframework.security.jwt.Jwt;
+
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Map;
+
+import static com.faforever.api.error.ApiExceptionWithCode.apiExceptionWithCode;
+import static org.hamcrest.Matchers.greaterThan;
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertThat;
+import static org.junit.Assert.fail;
+import static org.mockito.Matchers.any;
+import static org.mockito.Mockito.verify;
+import static org.mockito.Mockito.when;
+
+@RunWith(MockitoJUnitRunner.class)
+public class ClanServiceTest {
+  private ClanService instance;
+  @Rule
+  public ExpectedException expectedException = ExpectedException.none();
+
+  @Mock
+  private ClanRepository clanRepository;
+  @Mock
+  private PlayerRepository playerRepository;
+  @Mock
+  private FafApiProperties fafApiProperties;
+  @Mock
+  private JwtService jwtService;
+  @Mock
+  private AuthenticationService authenticationService;
+  @Mock
+  private ClanMembershipRepository clanMembershipRepository;
+
+  @Before
+  public void setUp() throws Exception {
+    instance = new ClanService(clanRepository, playerRepository, fafApiProperties, jwtService, authenticationService, clanMembershipRepository);
+  }
+
+  @Test
+  public void createClanWhereLeaderIsAllreadyInAClan() {
+    String clanName = "My cool Clan";
+    String tag = "123";
+    String description = "A cool clan for testing";
+    Player creator = new Player();
+    creator.setId(1);
+    creator.setClanMemberships(new ArrayList<>());
+    creator.getClanMemberships().add(new ClanMembership());
+    try {
+      instance.create(clanName, tag, description, creator);
+      fail();
+    } catch (ApiException e) {
+      assertThat(e, apiExceptionWithCode(ErrorCode.CLAN_CREATE_CREATOR_IS_IN_A_CLAN));
+    }
+    verify(clanRepository, Mockito.never()).save(any(Clan.class));
+  }
+
+  @Test
+  public void createClanSuccessful() {
+    String clanName = "My cool Clan";
+    String tag = "123";
+    String description = "A cool clan for testing";
+    Player creator = new Player();
+    creator.setId(1);
+    creator.setClanMemberships(new ArrayList<>());
+
+
+    instance.create(clanName, tag, description, creator);
+    ArgumentCaptor<Clan> clanCaptor = ArgumentCaptor.forClass(Clan.class);
+    verify(clanRepository, Mockito.times(1)).save(clanCaptor.capture());
+    assertEquals(clanName, clanCaptor.getValue().getName());
+    assertEquals(tag, clanCaptor.getValue().getTag());
+    assertEquals(description, clanCaptor.getValue().getDescription());
+    assertEquals(creator, clanCaptor.getValue().getLeader());
+    assertEquals(creator, clanCaptor.getValue().getFounder());
+    assertEquals(1, clanCaptor.getValue().getMemberships().size());
+    assertEquals(creator, clanCaptor.getValue().getMemberships().get(0).getPlayer());
+  }
+
+  @Test
+  public void generatePlayerInvitationTokenWithInvalidClan() throws IOException {
+    try {
```

Comment:
I think that either `new Player()` nor `setId()`  can throw an ApiException, therefore reduce the scope of the `try {}`